guppylang/definition/custom.py

```diff
@@ -261,12 +261,16 @@ def _setup(self, ctx: Context, node: AstNode, func: CustomFunctionDef) -> None:
         self.node = node
         self.func = func
 
-    @abstractmethod
     def check(self, args: list[ast.expr], ty: Type) -> tuple[ast.expr, Subst]:
         """Checks the return value against a given type.
 
         Returns a (possibly) transformed and annotated AST node for the call.
         """
+        from guppylang.checker.expr_checker import check_type_against
+
+        expr, res_ty = self.synthesize(args)
+        subst, _ = check_type_against(res_ty, ty, self.node)
+        return expr, subst
```

Comment:
Drive-by: Add a default implementation for `CustomCallChecker.check` that falls back to the synthesizing case if we can't make use of the incoming type information